### PR TITLE
{bio}[GCC/11.2.0] CIRCexplorer2 v2.3.8, Pysam v0.18.0, TopHat v2.1.0, pybedtools 0.8.2 w/ Python 2.7.18

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

Required to pass tests:
* https://github.com/easybuilders/easybuild-easyconfigs/pull/18843

### CODE REVIEW

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 16356` and update the PR to use `SYSTEM` instead.

File: easybuild/easyconfigs/t/TopHat/TopHat-2.1.0-gompi-2021b-Python-2.7.18.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+source_urls = ['http://ccb.jhu.edu/software/tophat/downloads/']
+sources = ['%(namelower)s-%(version)s.tar.gz']
+checksums = ['e7fa2b170408e785e4fafd0bccf4ecfab5151c480a2438dffa42183cfda71f99']
+
+builddependencies = [('Autoconf', '2.71')]
+
+dependencies = [
+    ('Python', '2.7.18'),
+    ('SAMtools', '1.14'),
+    ('Boost', '1.75.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/t/TopHat/TopHat-2.1.0-gompi-2021b-Python-2.7.18.eb:29]: @ItIsI-Orient The Boost 1.75.0 requirement is causing some "fallout" higher up in the dependency tree, so I wonder if patching TopHat to be compatible with Boost 1.77.0 is a viable approach. Did you look into that?

Comment (boegel) [reply easybuild/easyconfigs/t/TopHat/TopHat-2.1.0-gompi-2021b-Python-2.7.18.eb:29]: Could try with `'static': True` in `toolchainopts`, and then check the `tophat` binary with `ldd` to see if Boost gets linked statically...

Comment (jfgrimm) [reply easybuild/easyconfigs/t/TopHat/TopHat-2.1.0-gompi-2021b-Python-2.7.18.eb:29]: hmm, the `tophat` executable is actually not being dynamically linked regardless of the toolchainopt, only `segment_juncs` dynamically links against `Boost`

Comment (jfgrimm) [reply easybuild/easyconfigs/t/TopHat/TopHat-2.1.0-gompi-2021b-Python-2.7.18.eb:29]: adding `'static': True` to `toolchainopts` does result in a static `segment_juncs`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3108.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/946bdff6896f7434be017bcc6a1e7289 for a full test report.

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node4002.donphan.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 530.30.02, Python 3.6.8
See https://gist.github.com/ItIsI-Orient/2484e39a60730de13e7c292fc71eded0 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3157.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/db058840ac9c6656c0a72a8f3bf75ced for a full test report.

**Comment by boegel** [comment]
failed test report on `jsc-zen2` is because of download problem for `segemehl`, which is being fixed in #18878

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!

**Pull request merged**